src/Microsoft.AspNet.Server.Kestrel/Http/FrameOfT.cs

```diff
@@ -111,7 +112,7 @@ public class Frame<TContext> : Frame
                             _application.DisposeContext(context, _applicationException);
 
                             // If _requestAbort is set, the connection has already been closed.
-                            if (!_requestAborted)
+                            if (Volatile.Read(ref _requestAborted) == 0)
```

Comment:
What is the difference between having a `volatile int` and doing `Volatile.Read`?